app/helpers/proxmox_vm_attrs_helper.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+# Copyright 2018 Tristan Robert
+
+# This file is part of ForemanFogProxmox.
+
+# ForemanFogProxmox is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# ForemanFogProxmox is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with ForemanFogProxmox. If not, see <http://www.gnu.org/licenses/>.
+
+require 'fog/proxmox/helpers/disk_helper'
+require 'fog/proxmox/helpers/nic_helper'
+require 'fog/proxmox/helpers/cpu_helper'
+require 'foreman_fog_proxmox/value'
+require 'foreman_fog_proxmox/hash_collection'
+
+# Convert a foreman form server hash into a fog-proxmox server attributes hash
+module ProxmoxVmAttrsHelper
+  def object_to_attributes_hash(vms, from_profile)
+    param_scope = from_profile ? "compute_attribute[vm_attrs]" : "host[compute_attributes]"
+    vm_h = ActiveSupport::HashWithIndifferentAccess.new
+    keys = [:vmid, :node_id, :type, :pool]
+    main = vms.attributes.select { |key, _value| keys.include? key }
+    vms.config.all_attributes.each do |key, value|
+      vm_h.merge!({ key => { :name => "#{param_scope}[config_attributes][#{key}]", :value => value } }) unless keys.include? key
+    end
+    main.each do |key, value|
+      vm_h.merge!({ key => { :name => "#{param_scope}[#{key}]", :value => value } })
+    end
```

Comment:
```suggestion
    vms.config.all_attributes.each do |key, value|
      vm_h[key] = { :name => "#{param_scope}[config_attributes][#{key}]", :value => value } unless keys.include? key
    end
    main.each do |key, value|
      vm_h[key] = { :name => "#{param_scope}[#{key}]", :value => value } if keys.include? key
    end
```
I would have suggested using `transform_values()` to do it in one swoop, but apparently the block only gets the value and not the key (which we need for the `name`).

Reply:
added the changes